Neos.Flow/Classes/Http/RequestHandler.php

```diff
@@ -104,9 +105,13 @@ public function handleRequest()
 
         $this->boot();
         $this->resolveDependencies();
-        $this->addPoweredByHeader($response);
+        $response = $this->addPoweredByHeader($response);
+        $this->componentContext->replaceHttpResponse($response);
         if (isset($this->settings['http']['baseUri'])) {
-            $request->setBaseUri(new Uri($this->settings['http']['baseUri']));
+            $baseUri = new Uri($this->settings['http']['baseUri']);
+            $request = $request->withAttribute(Request::ATTRIBUTE_BASE_URI, $baseUri);
+            $this->componentContext->replaceHttpRequest($request);
+            $request->setBaseUri($baseUri);
```

Comment:
Needs further refactoring before I am confident to remove it.

Reply:
One of the main problems is that we have baseUri on the BaseRequest while Attributes only exist on ServerRequests, so we cannot replace baseUri with attributes on the BaseRequest. Any change to that would be a breaking one I guess, so I think we got to leave it like this until next major.

Reply:
Don't we have a ServerRequest here anyway? Otherwise we couldn't `withAttribute` either

Reply:
huh, why do you mean it#s created above? it gets an attribute, but the way the code is structured makes it at least technically breaking if we do not set the deprecated baseUri as well.

Reply:
Doesn't setBaseUri just set the attribute internally too?

Reply:
Yeah, good point, that is true only because we actually can know that we get a `Request` here and not something else. I was in a bit too deep and thought it may happen that we have an instance of `BaseRequest` (which also has the set/get baseUri methods) but has no attributes, so yes, rather pointless with the overriden baseuri methods in `Request`. Will change.